source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -275,9 +296,12 @@ Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& he
     const std::string escaped_redirect_uri =
         Http::Utility::PercentEncoding::encode(redirect_uri, ":/=&?");
 
+    const std::string escaped_auth_scopes =
+        Http::Utility::PercentEncoding::encode(absl::StrJoin(config_->authScopes(), " "), ":/=&? ");
```

Comment:
Maybe we should have config hold the encoded string instead of doing this encoding on every request?

Reply:
this makes sense, will work on it.